library/Witch/src/Juvix/Witch/CPSTranslation/Transform.hs

```diff
@@ -0,0 +1,410 @@
+{-# LANGUAGE RecordWildCards, DeriveFunctor, DeriveAnyClass #-}
+
+module Juvix.Witch.CPSTranslation.Transform where
+
+import Juvix.Library
+import qualified Juvix.Sexp as Sexp
+import qualified Juvix.Witch.CPSTranslation.SexpHelpers as Sexp
+import qualified Juvix.Sexp.Structure.EffectHandlerHelpers as Str
+import qualified Juvix.Sexp.Structure as Str
+import qualified Juvix.Sexp.Structure.CoreNamed as Str
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import qualified Juvix.Context.NameSpace as NameSpace
+import qualified Data.Unique as Uni
+import qualified Juvix.Context as Context
+import qualified Juvix.Context.Precedence as Pred
+
+-- | The CPS translation implemented here deals with higher-order continuations.
+-- | We work with continuations at the meta-language level (Haskell) and
+-- | source language (Juvix post-effects). For that, we convert the stack
+-- | between static and dynamic contexts.
+-- | Cryptic parts of the code are commented with a version that is more human-readable
+-- | and it appears above and horizontally-aligned with the code it explains.
+
+-- algorithm:
+-- put all handlers into context
+-- search for all vias
+-- for all vias
+--  - if current via contain any via, remove nested vias from main list
+--  - create a stack for via's handler
+--  - traverse over values and computations applying stacks as per rules
+-- end for vias
+-- remove all handlers from context
+
```

Comment:
Can we put this above the `Module ... where` bit?

Reply:
np 👍